app/controllers/courses_controller.rb

```diff
@@ -12,14 +12,14 @@ def edit
     end
 
     def update
-        if @course.update_attributes(params[:student])
+        if @course.update_attributes(params[:course])
             redirect_to @course, notice: 'Course was successfully updated.'
         else
+            flash[:alert] = "Course was not updated."
```

Comment:
This alert shouldn't be needed if you're re-rendering the edit form with validation failures shown in the form. No issues keeping it here though.